src/ros_main.cpp

```diff
@@ -134,22 +138,33 @@ int main(int argc, char **argv)
   auto rt_commander = factory.getCommander(rt_stream);
   vector<IConsumer<RTPacket> *> rt_vec{ &rt_pub };
 
-  TrajectoryFollower traj_follower(*rt_commander, local_ip, args.reverse_port, factory.isVersion3());
-
   INotifier *notifier(nullptr);
   ROSController *controller(nullptr);
   ActionServer *action_server(nullptr);
   if (args.use_ros_control)
   {
     LOG_INFO("ROS control enabled");
+    TrajectoryFollower traj_follower(*rt_commander, local_ip, args.reverse_port, factory.isVersion3());
     controller = new ROSController(*rt_commander, traj_follower, args.joint_names, args.max_vel_change, args.tcp_link);
```

Comment:
I have a suspicion this doesn't work due to the TrajectoryFollower being deallocated when it leaves the scope, which according to https://stackoverflow.com/questions/3097593/what-happens-when-c-reference-leaves-its-scope should be undefined behavior. 
However it seems my docker URSim image seems to have bit-rotted and can no longer launch so I'm unable to confirm it works, could you please confirm the ROSControl code path works? Merely starting with urX_ros_control.launch and letting it connect should should be enough see if it explodes or not.

If it works I'll make sure to merge ASAP

Reply:
I think you are completely right @Zagitta - even when it worked, it worked purely by chance and the behaviour could be different with different compilers or maybe even with different compiler options..

Indeed. In the previous version local variable initialisation worked fine because there was the spin() method below that kept the TrajectoryFollower object from destroying but when I moved it down the scope, it could have been destroyed and overwritten by another object. That's what happens wher you are mostly python programmer - there are no problems with out-of scope variables then. It's a pity though the compiler did not warn about it :). Regardless - I changed it now to allocate the TrajectoryFollower explicitely using new (consistently with the non-ros-control branch). Since this is all in main method, It's perfectly ok to not destroy it, I believe :). 

I  also have no setup to test it for ros control quickly, so maybe @psajecki  - you can help on Monday and try the new version with ros control as well :). That would have been helpful.